pkg/kmodule/kmodule_linux.go

```diff
@@ -103,6 +113,55 @@ func Delete(name string, flags uintptr) error {
 	return unix.DeleteModule(name, int(flags))
 }
 
+// parallelProbeDep loads a module and it's dependencies
+// Returns once module is loaded.
+func parallelProbeDep(deps depMap, modPath string, params string, opts ProbeOpts) error {
+	dep, present := deps[modPath]
```

Comment:
is there some guarantee that this is race-free?

Reply:
Since we load the entire depMap at once, it should be race-free if deps is read-only (and if modifying deps[modPath] is only done via its Once).